SU and Ingleby Codesquad_javascript_step_two

jeongyongsu5366 opened the pull request:
SU and Ingleby Codesquad_javascript_step_two

jeongyongsu5366 committed:
SU and Ingleby Codesquad_javascript_step_two

crongro commented on Figure/main.js:
```diff
@@ -0,0 +1,62 @@
+const circleArea = require('./module/figure/circleArea.js');
+const square = require('./module/figure/square.js');
+const trapezoid = require('./module/figure/trapezoid.js');
+const cylinder = require('./module/figure/cylinder.js');
```
모듈이 너무 많네요 ㅎ
다 비슷한 역할임으로 하나의 모듈로 넓이계산하는 녀석들을 합쳐주는 건 어때요?

crongro commented on Figure/main.js:
```diff
@@ -0,0 +1,62 @@
+const rl = readline.createInterface(process.stdin, process.stdout);
+
+var recursiveAsyncReadLine = function() {
+    rl.question('Which figure area do you want to know? 1. Circle 2. Square 3. Trapezoid 4. Cylinder:  ', function(answer) {
```
오 이런 사용자  친화적인 기능!!
💯

crongro commented on Figure/module/figure/cylinder.js:
```diff
@@ -0,0 +1,11 @@
+module.exports = (radius, height) => {
+        try {
```
try/catch사용법을 잘 익히고 계시네요.
잘했습니다.

crongro commented on Figure/main.js:
```diff
@@ -0,0 +1,62 @@
+//const myModule = require('./module/figure');
```
figure/index는 쓰려다 안썼군요.
음 뭐가 더 좋은걸까요? 생각을 댓글로 좀 남겨주세요.

crongro commented on Figure/module/figure/trapezoid.js:
```diff
@@ -0,0 +1,11 @@
+module.exports = (Bbase, Tbase, height) => {
+    try {
+        if((isNaN(Bbase) || isNaN(Tbase) || isNaN(height))) {
```
약간 중복코드같아요.  조건문안에 같은 3가지를 한번에 하는 방법도 있을 거 같아요.
한번 찾아보세요.

crongro reviewed (changes requested):
사용자 친화적인 기능만드느라 수고하셨네요. 잘하셨어요.

몇개 리팩토링 부탁드리고, 질문에 설명도 부탁드립니다.

jeongyongsu5366 committed:
1. We have gathered a variety of similar methods in a module called
figureArea.js.

2. We have created the method that wehn occuring type error, return to
the very beginning of the function recursiveAsynceFigureArea using the
concept of recursion.

3. We have maximized the reusability of the methods that we made as
possible as we can :D

crongro commented on Figure/module/figure/figureArea.js:
```diff
@@ -0,0 +1,51 @@
+  for (let i = 0; i < args.length; i++) {
+      sum += args[i];
+  }
+  try {
```
사실 이코드는, 

if(isNaN(sum)) console.log("Invalid Input!! Start Again!!\n"); 

이렇게만 해도 똑같긴 하네요

crongro commented on Figure/main.js:
```diff
@@ -43,20 +39,33 @@ var recursiveAsyncReadLine = function() {
+});
+
+
+function errorHandler(result) {
```
중복 부분을 없애려는 시도가 좋네요.

crongro commented on Figure/main.js:
```diff
@@ -43,20 +39,33 @@ var recursiveAsyncReadLine = function() {
-            console.log(cylinder(radHei[0], radHei[1]));
-                rl.close();
+            const resultOfCylinder = figureArea.cylinder(Number(radHei[0]), Number(radHei[1]));
+            errorHandler(resultOfCylinder);
```
동사+명사가 좀더 일반적인 네이밍방법이죠.
handleError();

crongro reviewed (approved):
네 수고하셨어요.

머지할게요~
